src/fasta.rs

```diff
@@ -203,27 +213,32 @@ enum ParserState {
     InRecord {
         start_line_number: usize,
         header: String,
-        contents: String,
+        contents: T,
     },
 }
 
-impl ParserState {
+type ParseLineResult<T> = Result<
+    (ParserState<T>, Option<FastaRecord<T>>),
+    Located<FastaParseError<<T as FastaContent>::Err>>,
+>;
+
+impl<T: FastaContent> ParserState<T> {
     /// Pump the state machine and maybe emit a record, if this line completes a record, along with
     /// a new state
     fn advance_line(
         self,
         settings: &FastaParseSettings,
         line: &str,
         line_number: usize,
-    ) -> (Self, Option<FastaRecord<String>>) {
+    ) -> ParseLineResult<T> {
         let new_header = try_parse_header(line);
-        match (self, new_header) {
+        let (new_state, record) = match (self, new_header) {
             // start of file, and we have a header line => start a new record,
             // maybe emiting the preceding content as a headerless record depending
             // on parse settings
             (ParserState::StartOfFile { contents }, Some(new_header)) => {
                 // don't emit if the settings don't want it, or if the record would just be whitespace
-                let record = if settings.allow_preceding_comment || contents.trim().is_empty() {
+                let record = if settings.allow_preceding_comment || contents.is_empty() {
```

Comment:
I think this is a behavior change? I believe this `trim()` is important for allowing whitespace at the beginning of a file even in no-comment mode:

```

> header
...contents...
```

should result in a single record starting on line 2 regardless of parse settings.

Reply:
We should probably have a test for this

Reply:
I renamed it to `is_blank()`.